guillotina/json/deserialize_content.py

```diff
@@ -72,21 +72,22 @@ def __init__(self, context, request):
 
         factory = get_cached_factory(self.context.type_name)
         main_schema = factory.schema
-        await self.set_schema(
+        changed = await self.set_schema(
             main_schema, self.context, data, errors, validate_all, False)
 
         if errors and not ignore_errors:
             raise DeserializationError(errors)
 
-        self.context.register()
+        if changed:
+            self.context.register()
 
         return self.context
 
     async def set_schema(
             self, schema, obj, data, errors,
-            validate_all=False, behavior=False):
+            validate_all=False, behavior=False) -> bool:
         write_permissions = merged_tagged_value_dict(schema, write_permission.key)
-
+        changed = False
```

Comment:
It's just a small concern to have that flag named `schema_changed` instead of `changed`. As it seems to give more context about the flag.